c_glib/arrow-glib/basic-array.cpp

```diff
@@ -83,6 +83,15 @@ garrow_primitive_array_new(GArrowDataType *data_type,
   return garrow_array_new_raw(&arrow_array);
 };
 
+template <typename T>
+typename T::c_type
+garrow_numeric_array_get_sum_value_raw(std::shared_ptr<arrow::Scalar> scalar)
+{
+  using ScalarType = typename arrow::TypeTraits<T>::ScalarType;
+  auto arrow_numeric_scalar = std::dynamic_pointer_cast<ScalarType>(scalar);
+  return arrow_numeric_scalar->value;
+};
+
```

Comment:
Can we create a template function that can be used by all `sum` functions?

```diff
diff --git a/c_glib/arrow-glib/basic-array.cpp b/c_glib/arrow-glib/basic-array.cpp
index deedc0fe..4db6a564 100644
--- a/c_glib/arrow-glib/basic-array.cpp
+++ b/c_glib/arrow-glib/basic-array.cpp
@@ -92,6 +92,28 @@ garrow_numeric_array_get_sum_value_raw(std::shared_ptr<arrow::Scalar> scalar)
   return arrow_numeric_scalar->value;
 };
 
+template <typename ArrowType, typename GArrowArrayType>
+typename ArrowType::c_type
+garrow_numeric_array_sum(GArrowArrayType array,
+                         GError **error,
+                         const gchar *tag,
+                         typename ArrowType::c_type default_value)
+{
+  auto arrow_array = garrow_array_get_raw(GARROW_ARRAY(array));
+  auto memory_pool = arrow::default_memory_pool();
+  arrow::compute::FunctionContext context(memory_pool);
+  arrow::compute::Datum sum_datum;
+  auto status = arrow::compute::Sum(&context, arrow_array, &sum_datum);
+  if (garrow_error_check(error, status, tag)) {
+    using ScalarType = typename arrow::TypeTraits<ArrowType>::ScalarType;
+    auto arrow_numeric_scalar =
+      std::dynamic_pointer_cast<ScalarType>(sum_datum.scalar());
+    return arrow_numeric_scalar->value;
+  } else {
+    return default_value;
+  }
+}
+
 G_BEGIN_DECLS
 
 /**
@@ -1062,16 +1084,10 @@ gint8
 garrow_int8_array_sum(GArrowInt8Array *array,
                       GError **error)
 {
-  auto arrow_array = garrow_array_get_raw(GARROW_ARRAY(array));
-  auto memory_pool = arrow::default_memory_pool();
-  arrow::compute::FunctionContext context(memory_pool);
-  arrow::compute::Datum sum_datum;
-  auto status = arrow::compute::Sum(&context, arrow_array, &sum_datum);
-  if (garrow_error_check(error, status, "[int8-array][sum]")) {
-    return garrow_numeric_array_get_sum_value_raw<arrow::Int64Type>(sum_datum.scalar());
-  } else {
-    return 0;
-  }
+  return garrow_numeric_array_sum<arrow::Int64Type>(array,
+                                                    error,
+                                                    "[int8-array][sum]",
+                                                    0);
 }
```

Reply:
Thanks! I've updated a template function.